### PR TITLE
Improve API for use by language server

### PR DESCRIPTION
**Pull request by dsainati1**
Part of https://github.com/onflow/cadence/issues/1269

Exposes some functions and types for use by the language server
______

- [X] Targeted PR against `master` branch
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/cadence-lint/blob/master/CONTRIBUTING.md#styleguides)
- [X] Updated relevant documentation 
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [X] Added appropriate labels

### CODE REVIEW

File: linter/linters.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+package linter
+
+import (
+	"github.com/onflow/cadence-lint/analyzers"
+	"github.com/onflow/cadence/tools/analysis"
+)
+
+var LinterAnalyzers = []*analysis.Analyzer{
```
Comment (turbolent) [inline linter/linters.go:26]: BTW, there's also `Analyzers` defined already: https://github.com/onflow/cadence-lint/blob/062942ee8b35ecb2e8961aa93eab4ed4b8f32883/analyzers/analyzer.go#L28

File: analyzers/number_function_argument_analyzer.go [hunk 2]
```diff
@@ -33,10 +31,11 @@ func ReplacementHint(
 	r ast.Range,
 ) *analysis.Diagnostic {
 	return &analysis.Diagnostic{
-		Location: location,
-		Range:    r,
-		Category: "lint",
-		Message:  fmt.Sprintf("consider replacing with: `%s`", expr),
+		Location:         location,
+		Range:            r,
+		Category:         "replacement-hint",
```
Comment (SupunS) [inline analyzers/number_function_argument_analyzer.go:36]: What are all possible lint categories? Does each type of hint belong to its own category?

Perhaps define enum-like constants for 'category'.

**Review by SupunS — APPROVED** [review]
Looks good. Just one question on the hint-categories

File: analyzers/analyzer.go [hunk 3]
```diff
@@ -25,6 +25,13 @@ import (
 	"github.com/onflow/cadence/tools/analysis"
 )
 
+const (
+	ReplacementCategory     string = "replacement-hint"
+	RemovalCategory         string = "removal-hint"
+	UpdateCategory          string = "update recommended"
+	UnnecessaryCastCategory string = "unnecessary-cast-hint"
```
[the remark this answers is not in the document]
Comment (dsainati1) [reply analyzers/analyzer.go:32]: Yeah, in https://github.com/onflow/cadence/pull/1801 we use `RemovalHint` to suggest specific code to be removed in the language server. This is theoretically behavior we could add for the unnecessary casts as well, but the goal with these PRs is just to replicate the old behavior with the new API. We could definitely do a follow-up PR to make `UnnecessaryCast` into a `Removal` hint.

**Pull request merged**